core/src/swarm.rs

```diff
@@ -164,12 +164,18 @@ where TBehaviour: NetworkBehaviour<TTopology>,
     }
 
     /// Tries to reach the given peer using the elements in the topology.
+    /// Silently returns if address was associated with the `Self`
     ///
     /// Has no effect if we are already connected to that peer, or if no address is known for the
     /// peer.
     #[inline]
     pub fn dial(me: &mut Self, peer_id: PeerId) {
         let addrs = me.topology.addresses_of_peer(&peer_id);
+        for a in addrs.iter() {
```

Comment:
Maybe use `any`? (untested!)
```suggestion
       let is_dialling_self = addrs.iter().any(|addr| {
            me.listened_addrs.iter().any(|my_addr| my_addr == addr)
        })
        if is_dialling_self {
            trace!("can't dial self!");
            return;
         }
```

Reply:
Yeah, didn't notice that function, thanks!